zebra-state/src/service.rs

```diff
@@ -803,6 +803,91 @@ impl ReadStateService {
     fn latest_non_finalized_state(&self) -> NonFinalizedState {
         self.non_finalized_state_receiver.cloned_watch_data()
     }
+
+    /// Check the contextual validity of a block in the best chain
+    #[cfg(feature = "getblocktemplate-rpcs")]
+    #[allow(clippy::unwrap_in_result)]
+    #[tracing::instrument(level = "debug", skip_all)]
+    fn check_best_chain_contextual_validity(
+        &self,
+        prepared: &PreparedBlock,
+    ) -> Result<(), crate::ValidateContextError> {
```

Comment:
This seems like it is a duplicate of existing code that implements the consensus rules.

So it could easily go out of sync as we change the rules. (Or be called accidentally from somewhere we want to check the full set of block rules before we commit.)

Is there a way to re-use the existing consensus rule checks instead?

For example, we could:
1. Create a deep cloned non-finalized state, which only contains the best chain (or nothing, if there is no best chain)
2. Call `commit_block()` / `commit_new_chain()` on that non-finalized state (this automatically deep clones the best chain if its `Arc` is still shared)
3. Drop the modified finalized state clone, which drops the modified chain
4. Add an argument to `commit_block()` / `commit_new_chain()` that skips the metrics updates, because the block isn't actually committed